beets/ui/__init__.py

```diff
@@ -367,6 +367,31 @@ def input_yn(prompt, require=False):
     return sel == 'y'
 
 
+def input_select_items(prompt, items, rep):
+    """Prompts the user to use all, none or some of the items
+    Will return the list of items the user selected
+    prompt: prompt to use for all and for selective choice
+    items: full list of items
+    rep: function which represents an item to the user
+         is called with the item as argument
+         function is responsive for newline at input
+    """
+    out_items = []
+    choice = input_options(
+        ('y', 'n', 's'), False,
+        '%s (Yes/No/Selective)?' % prompt)
+    print()
+    if choice == 'y':
+        out_items = items
+    elif choice == 's':
+        for item in items:
+            rep(item)
+            if input_yn('%s (y/n)?' % prompt, True):
```

Comment:
Sorry, that was confusingly put. Here's what I mean. Say you get a prompt like this:

```
Really do something to 1000 albums? (Yes/No/Select) s
Do something to album 1? n
Do something to album 2? n
...
```

If I accidentally hit "s" when I meant "n", it will be a long time before I can counteract my mistake. I'll have to hit "n" all day or find some other way to kill the process.

The same thing happens if I accidentally hit "y" for one of the thousand prompts and want to start over.

In both cases, it would be useful to have some action I could take to cancel the whole process at one of the individual prompts. That would case the `input_select_items` function to return an empty list, indicating that no action should be taken.